api/features/multivariate/models.py

```diff
@@ -59,6 +61,20 @@ def create_multivariate_feature_state_values(self):
                 percentage_allocation=self.default_percentage_allocation,
             )
 
+    @hook(AFTER_CREATE)
+    def make_feature_multivariate(self):
+        # We can't do this check using `when` and `is_not`
```

Comment:
This comment isn't _super_ clear. Is there some piece of documentation that we can link to? I assume it's related to triggering across the join? So, something in [these docs](https://rsinger86.github.io/django-lifecycle/fk_changes/) might be the issue?

Reply:
It's not really an issue, though - I don't think. It's just the way that this works, by capturing the state on __init__

Reply:
But, surely that's the case with the code we've got here as well, right? It's always going to check across the join?

Reply:
No, in our case it will only happen after_create

Reply:
Ah, interesting. Then yeah, django-lifecycle should definitely optimise that! I think the comment could be clearer then though to explicitly state _why_ we're doing it: 

```python
# Handle the check on feature.type in the method itself to ensure this is
# only performed after create. Using `when` and `is_not` means 
# LifecycleModel.__init__() queries for feature on every object init. 
```